frontend/src/pages/projects/screens/detail/storage/ClusterStorageTableRow.tsx

```diff
@@ -0,0 +1,292 @@
+    () => getInUseMountPaths(obj.name, availableNotebooks.notebooks, existingPvc?.metadata.name),
+    [availableNotebooks.notebooks, existingPvc?.metadata.name, obj.name],
+  );
+
+  React.useEffect(() => {
+    if (!obj.existingPvc && !obj.isUpdatedValue) {
+      const defaultPathValue = clusterStorageName
+        ? `${MOUNT_PATH_PREFIX}${clusterStorageName.toLowerCase().replace(/\s+/g, '-')}-${
+            obj.newRowId ?? 1
+          }`
+        : '';
+      addNotebookData(
+        undefined,
+        clusterStorageName
+          ? {
+              value: defaultPathValue,
+              error: validateClusterMountPath(defaultPathValue, inUseMountPaths),
+            }
+          : undefined,
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [clusterStorageName, inUseMountPaths, obj.existingPvc, obj.isUpdatedValue, obj.newRowId]);
+
+  const suffix = mountPathSuffix(obj.mountPath.value);
+  const format = mountPathFormat(obj.mountPath.value);
+
+  const pathFormatOptions = [
+    {
+      type: MountPathFormat.STANDARD,
+      description: `Standard paths that begins with ${MOUNT_PATH_PREFIX} are visible in JupyterLab file browser.`,
+    },
+    {
+      type: MountPathFormat.CUSTOM,
+      description: `Custom paths that do not begin with ${MOUNT_PATH_PREFIX} are not visible in the JupyterLab file browser.`,
+    },
+  ];
+
+  const selectOptions = React.useMemo(
+    () =>
+      availableNotebooks.notebooks.map((notebook) => ({
+        value: notebook.metadata.name,
+        content: notebook.metadata.name,
+      })),
+    [availableNotebooks.notebooks],
+  );
+
+  const validateAndUpdate = React.useCallback(
+    (value: string, newFormat?: 'Standard' | 'Custom') => {
+      const prefix = (newFormat ?? format) === MountPathFormat.STANDARD ? MOUNT_PATH_PREFIX : '/';
+      const newValue = `${prefix}${value}`;
+
+      const notebook = availableNotebooks.notebooks.find(
+        (currentNotebook) => currentNotebook.metadata.name === obj.name,
+      );
+      const existingPath = getNotebookPVCMountPathMap(notebook)[existingPvc?.metadata.name ?? ''];
+      const error = validateClusterMountPath(newValue, inUseMountPaths) || '';
+
+      const isSamePvcPath = obj.existingPvc && newValue === existingPath;
+      addNotebookData(obj.name, { value: newValue, error }, !isSamePvcPath);
+    },
+    [
+      addNotebookData,
+      availableNotebooks.notebooks,
+      existingPvc?.metadata.name,
+      format,
+      inUseMountPaths,
+      obj.existingPvc,
+      obj.name,
+    ],
+  );
+
+  const handleFormatChange = (newFormat: 'Standard' | 'Custom') => {
+    validateAndUpdate(suffix, newFormat);
+  };
+
+  let placeholderText: string;
+
+  if (!availableNotebooks.loaded) {
+    placeholderText = 'Loading workbenches';
+  } else if (selectOptions.length === 0) {
+    placeholderText = 'No existing workbench available';
+  } else {
+    placeholderText = 'Select a workbench';
+  }
```

Comment:
This can be refactored to useMemo

```
const placeholderText = React.useMemo(() => {
 -----
}, [availableNotebooks.loaded, selectOptions.length]);
```

Reply:
@dpanshug , I’ve seen this pattern used in several places throughout our codebase. However, I’m open to refactoring it to useMemo.

Reply:
this should not be use memo, it is not really optimizing anything. useMemo should be reserved for when we have things that are at least O(n), but honestly JS does a good job with that and would could get away without memoizing that.